go/tasks/plugins/k8s/spark/spark.go

```diff
@@ -137,12 +137,24 @@ func (sparkResourceHandler) BuildResource(ctx context.Context, taskCtx pluginsCo
 	}
 
 	// Set pod limits.
-	if sparkConfig["spark.kubernetes.driver.limit.cores"] == "" && sparkConfig["spark.driver.cores"] != "" {
-		sparkConfig["spark.kubernetes.driver.limit.cores"] = sparkConfig["spark.driver.cores"]
+	if sparkConfig["spark.kubernetes.driver.limit.cores"] == "" {
```

Comment:
nit:
```suggestion
	if len(sparkConfig["spark.kubernetes.driver.limit.cores"]) == 0 {
```

Reply:
replaced the check everywhere with len()